src/data/shortlist.cpp

```diff
@@ -0,0 +1,396 @@
+#include "shortlist.h"
+#include <queue>
+
+namespace marian {
+namespace data {
+
+bool isBinaryShortlist(const std::string& fileName){
+  uint64_t magic;
+  io::InputFileStream in(fileName);
+  in.read((char*)(&magic), sizeof(magic));
+  return in && (magic == BINARY_SHORTLIST_MAGIC);
+}
+
+void LexicalShortlistGenerator::load(const std::string& fname) {
```

Comment:
Why is the preexisting lexical shortlist being added?  Is it in some other file?